src/dc_receive_imf.rs

```diff
@@ -1332,6 +1359,19 @@ async fn create_or_lookup_group(
             });
     };
 
+    // check, if we have a chat with this group ID
+    let (mut chat_id, _, _blocked) = chat::get_chat_id_by_grpid(context, &grpid)
+        .await
+        .unwrap_or((ChatId::new(0), false, Blocked::Not));
+
+    // For chat messages, we don't have to guess (is_*probably*_private_reply()) but we know for sure that
+    // they belong to the group because of the Chat-Group-Id or Message-Id header
```

Comment:
Maybe check for `Chat-Group-Id` directly instead of `Chat-Version`? See my comment above about possibly removing `Chat-Group-Id` for ad-hoc groups.

Reply:
Writing things down not to have to keep things in mind, conclusion is below.

Context is: I added this `!mime_parser.has_chat_version() &&` to make two-member-chat-groups work: 

In `is_probably_private_reply` it says:

    // There is one exception: Classical MUA replies to two-member groups
    // should be assigned to the group chat. We restrict this exception to classical emails, as chat-group-messages
    // contain a Chat-Group-Id header and can be sorted into the correct chat this way.

Now, here we are in `create_or_lookup_group()`, and here we have to look at the Chat-Group-Id header and sort messages into the correct chat this way. Specifically, this is about chat messages to two-member-groups.

---

...buut `try_getting_grpid()` does not only look at Chat-Group-Id, but also at Message-Id, with the comment "Useful for undecipherable messages sent to known group.". I chose `!mime_parser.has_chat_version()` as it is also used in `try_getting_grpid()`: Obviously we are here because `try_getting_grpid()` returned Some. So, if `has_chat_version()` is false, we can be sure that the grpid was extracted from In-Reply-To or References (as classical emails don't include a grpid in Chat-Group-Id or Message-Id). And if `has_chat_version()` is true, we can be sure that it was extracted from Chat-Group-Id or Message-Id, as `try_getting_groupid()` would return before extracting it from somewhere else.

**Conclusion:**

If we only look at the Chat-Group-Id header here, undecipherable chat messages to two member groups will be sorted to the private chat with the sender instead of the group chat.

As we look at Chat-Version header here, there should be only be the risk that private replies to two member ad-hoc groups are sorted to the group chat instead of the private chat. Buuut there are no two-member ad-hoc groups, anyway.

Options I see for improvement:
- Leave things as they are, but improve the comment to explain more context, because I had to think really long to remember why I had even put this there
- Let `try_getting_grpid()` also return a bool which we can then call `grpid_is_from_parent` or similar, which is true if we extracted the grpid from `InReplyTo` or `References`. Then we can do `if grpid_is_from_parent && is_probably_private_reply(context, to_ids, mime_parser, chat_id, from_id).await?`. No semantic change, but the relationship between `try_getting_grpid()` and this check here is no longer disguised.